### PR TITLE
[ScalarizeMaskedMemIntrin] Use pointer alignment from pointer of masked.compressstore/expandload.

### PR DESCRIPTION
**Pull request by yetingk**
Previously we used Align(1) for all scalarized load/stores from masked.compressstore/expandload.
For targets not supporting unaligned accesses, it make backend need to split
aligned large width loads/stores to byte loads/stores.
To solve this performance issue, this patch preserves the alignment of base
pointer after scalarizing.

### CODE REVIEW

File: llvm/lib/Transforms/Scalar/ScalarizeMaskedMemIntrin.cpp [hunk 1]
```diff
@@ -659,7 +660,7 @@ static void scalarizeMaskedExpandLoad(const DataLayout &DL, CallInst *CI,
       } else {
         Value *NewPtr =
             Builder.CreateConstInBoundsGEP1_32(EltTy, Ptr, MemIndex);
-        InsertElt = Builder.CreateAlignedLoad(EltTy, NewPtr, Align(1),
+        InsertElt = Builder.CreateAlignedLoad(EltTy, NewPtr, Alignment,
```
Comment (topperc) [inline llvm/lib/Transforms/Scalar/ScalarizeMaskedMemIntrin.cpp:663]: We need to use `commonAlignment(Alignment, EltTy->getSizeInBits() / 8)`

File: llvm/test/CodeGen/RISCV/rvv/fixed-vectors-compressstore-int.ll [hunk 2]
```diff
@@ -0,0 +1,986 @@
...
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    vsetvli a1, zero, e8, mf8, ta, ma
+; CHECK-NEXT:    vfirst.m a1, v0
+; CHECK-NEXT:    bnez a1, .LBB0_2
+; CHECK-NEXT:  # %bb.1: # %cond.store
+; CHECK-NEXT:    vsetivli zero, 1, e8, mf8, ta, ma
+; CHECK-NEXT:    vse8.v v8, (a0)
+; CHECK-NEXT:  .LBB0_2: # %else
+; CHECK-NEXT:    ret
+  call void @llvm.masked.compressstore.v1i8(<1 x i8> %v, ptr %base, <1 x i1> %mask)
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rvv/fixed-vectors-compressstore-int.ll:17]: The alignment attribute should be on parameter in the intrinsic call. See for example, `unaligned_vpload_nxv1i64_allones_mask` in vpload.ll

File: llvm/lib/Transforms/Scalar/ScalarizeMaskedMemIntrin.cpp [hunk 3]
```diff
@@ -627,6 +627,7 @@ static void scalarizeMaskedExpandLoad(const DataLayout &DL, CallInst *CI,
   Value *Ptr = CI->getArgOperand(0);
   Value *Mask = CI->getArgOperand(1);
   Value *PassThru = CI->getArgOperand(2);
+  Align Alignment = Ptr->getPointerAlignment(DL);
```
Comment (topperc) [inline llvm/lib/Transforms/Scalar/ScalarizeMaskedMemIntrin.cpp:630]: I think we should be using CI->getParamAlign here.

**Review by preames — APPROVED** [review]
LGTM

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**